test-resources/headless-core-stub/lambdas/callback/src/callback-handler.ts

```diff
@@ -1,12 +1,91 @@
-import type { LambdaInterface } from "@aws-lambda-powertools/commons/types";
+import { LambdaInterface } from "@aws-lambda-powertools/commons/types";
+import { APIGatewayProxyEvent, APIGatewayProxyResult, Context } from "aws-lambda";
+import { Logger } from "@aws-lambda-powertools/logger";
+import { ConfigurationHelper } from "./services/configuration-helper";
+import { CallBackService } from "./services/callback-service";
+import { generatePrivateJwtParams } from "./services/private-key-jwt-helper";
+import { JWK } from "jose";
 
+const sessionTableName = process.env.SESSION_TABLE || "session-common-cri-api";
+const configurationHelper = new ConfigurationHelper();
+const callbackService = new CallBackService();
+const logger = new Logger();
 export class CallbackLambdaHandler implements LambdaInterface {
-    async handler(): Promise<{ status: string; body: string }> {
-        return {
-            status: "200",
-            body: "Hello callback",
-        };
+    async handler(event: APIGatewayProxyEvent, _context: Context): Promise<APIGatewayProxyResult> {
+        try {
+            const authorizationCode = event.queryStringParameters?.authorizationCode as string;
+            logger.info({ message: "Received authorizationCode", authorizationCode });
+
+            const sessionItem = await callbackService.getSessionByAuthorizationCode(
+                sessionTableName,
+                authorizationCode,
+            );
+            logger.info({ message: "Fetching session item...", ...sessionItem });
+
+            const ssmParameters = await configurationHelper.getParameters(sessionItem.clientId);
+            const { privateSigningKey, ...filteredParams } = ssmParameters;
+
+            logger.info({ message: "Fetching SSM parameters", ...filteredParams });
+
+            const privateJwtKey = JSON.parse(privateSigningKey) as JWK;
+            const audience = ssmParameters["audience"];
+
+            logger.info({ message: "Generating private JWT parameters..." });
+            const privateJwtParams = await generatePrivateJwtParams(
+                sessionItem.clientId,
+                authorizationCode,
+                sessionItem.redirectUri,
+                privateJwtKey,
+                audience,
+            );
+
+            const audienceApi = this.formatAudience(audience);
+            logger.info({ message: "Using Audience", audienceApi });
+
+            const tokenEndpoint = `${audienceApi}/token`;
+            logger.info({ message: "Calling token endpoint", tokenEndpoint, privateJwtParams });
+            const tokenResponse = await callbackService.getToken(tokenEndpoint, privateJwtParams);
+
+            if (!tokenResponse.ok) {
+                const tokenResponseBody = await tokenResponse.text();
+                const status = tokenResponse.status;
+                logger.error({ message: "Request to endpoint failed", tokenEndpoint, status, tokenResponseBody });
```

Comment:
can we label the fields? e.g. 
`{ message: "Request to endpoint failed", endpoint: tokenEndpoint, statusCode: status, response: tokenResponseBody }`

Reply:
Similarly for the other log messages. Unless they translate correctly in CloudWatch.